[RS-2421] Add doc for configuring additional GatewayClasses

nelljerram opened the pull request:
FTR, it was eventually agreed for this PR to add doc about configuring additional GatewayClasses to the OSS docs, but not to Enterprise - to avoid any possible suggestion that customer-configured GatewayClasses would be supported.

There was an additional piece of extra doc talking about possible customizations via the GatewayAPI CR.  This is supported, and has therefore been retained for both OSS and Enterprise.

nelljerram committed:
Add doc for configuring additional GatewayClasses

nelljerram committed:
Add a section about available customization fields

ctauchen commented on calico/networking/gateway-api.mdx:
```diff
@@ -416,3 +418,60 @@ EOF
+
+To do this:
+- Extract the YAML for the EnvoyProxy and GatewayClass objects that the Tigera operator provisions.
+- Change the EnvoyProxy YAML according to your needs, and create that modified EnvoyProxy object with a different name (than "envoy-proxy-config").
```
```suggestion
- Change the EnvoyProxy YAML according to your needs, and rename the modified EnvoyProxy object (not `envoy-proxy-config.yaml`).
```
For clarity.

nelljerram replied:
I'm afraid this isn't a correct suggestion, because it really is creating a new object, not renaming an existing one.

ctauchen commented on calico/networking/gateway-api.mdx:
```diff
@@ -416,3 +418,60 @@ EOF
+To do this:
+- Extract the YAML for the EnvoyProxy and GatewayClass objects that the Tigera operator provisions.
+- Change the EnvoyProxy YAML according to your needs, and create that modified EnvoyProxy object with a different name (than "envoy-proxy-config").
+- Change the GatewayClass YAML to reference the name of the modified EnvoyProxy object, and create that modified GatewayClass with a different name (than "tigera-gateway-class").
```
Change as above.

ctauchen replied:
Also change "filename" throughout to ``filename``.

nelljerram replied:
First point - as in my previous response.

Second point - actually these are resource names, not file names.  But OK, I think your suggestion still makes sense.

ctauchen reviewed (commented):
A few comments, otherwise LGTM. 

I presume this needs to be copied elsewhere?

nelljerram committed:
Use code markup for resource names instead of double quotes

nelljerram committed:
Copy same material to Enterprise docs

The only difference is that the GatewayAPI resource is named "tigera-secure" in
Enterprise, and "default" in OSS.

nelljerram committed:
Merge remote-tracking branch 'origin/main' into gateway-api-objects

nelljerram committed:
Warn about custom GatewayClasses not being supported in Enterprise

ctauchen commented:
@nelljerram Considering Ash's comments, let's remove the changes from CE and CC and merge the OSS changes.

nelljerram committed:
Merge remote-tracking branch 'origin/main' into gateway-api-objects

nelljerram committed:
Don't document configuring additional GatewayClasses for Enterprise
